internal/files/repository_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package files
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRepositoryGitDirectory(t *testing.T) {
+	cases := []struct {
+		name          string
+		createGit     bool
+		repoDir       string
+		expectedError string
+		valid         bool
+	}{
+		{
+			name:          "git folder present",
+			createGit:     true,
+			repoDir:       t.TempDir(),
+			expectedError: "",
+			valid:         true,
+		},
+		{
+			name:          "no git folder",
+			createGit:     false,
+			repoDir:       t.TempDir(),
+			expectedError: "file does not exist",
+			valid:         false,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			gitWorktreeFile := filepath.Join(c.repoDir, ".git")
+			otherDir := filepath.Join(c.repoDir, "other")
+
+			if c.createGit {
+				err := os.MkdirAll(gitWorktreeFile, 0o755)
+				require.NoError(t, err)
+			}
+			err := os.MkdirAll(otherDir, 0o755)
+			require.NoError(t, err)
+
+			err = os.Chdir(otherDir)
+			require.NoError(t, err)
+
+			dir, err := FindRepositoryRootDirectory()
```

Comment:
Changing cwd is a bit dangerous, it changes cwd in the whole process and might affect other tests. It is also probably the reason why this fails in Windows, when the test finishes, the temporary directory that should be removed is the current working directory.
Maybe we can define an internal function that looks for the root directory under a given path, and test this function.
It would be something like this:
```
func FindRepositoryRootDirectory() (string, error) {
	workDir, err := os.Getwd()
	if err != nil {
		return "", fmt.Errorf("locating working directory failed: %w", err)
	}
	return findRepositoryRootDirectory(workDir)
}

func findRepositoryRootDirectory(workDir string) (string, error) {
	...
```
And then here:
```suggestion
			dir, err := findRepositoryRootDirectory(otherDir)
```
We wouldn't be testing the part about getting the local dir, but I think this is fine enough.

Reply:
> Maybe we can define an internal function that looks for the root directory under a given path, and test this function.

I'll try with this, thanks!